### PR TITLE
Add test to print device information

### PR DESCRIPTION
**Pull request by Rbiessy**
Add a test to print device information. This is the only solution I found to make this work while still using `ctest`.

## Checklist

Tick if relevant:

* [x] New files have a copyright
* [N/A] New headers have an include guards
* [N/A] API is documented with Doxygen
* [N/A] New functionalities are tested
* [x] Tests pass locally
* [x] Files are clang-formatted

### CODE REVIEW

File: test/unit_test/print_device.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+#include "sycl_utils.hpp"
+
+/**
+ * Print information of the device selected by the default selector.
+ * Use this as a test to print the information once when using ctest.
+ */
+void print_device() {
+  namespace info = sycl::info::device;
+  sycl::queue queue;
+  sycl::device dev = queue.get_device();
```
Comment (hjabird) [inline test/unit_test/print_device.cpp:32]:
```suggestion
  sycl::device dev();
```

**Review by hjabird — APPROVED** [review]
LGTM

**Review by hjabird — DISMISSED** [review]

**Review by AD2605 — APPROVED** [review]

File: test/unit_test/print_device.cpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+#include <gtest/gtest.h>
+
+#include "sycl_utils.hpp"
+
+/**
+ * Print information of the device selected by the default selector.
+ * Use this as a test to print the information once when using ctest.
+ */
+void print_device() {
```
Comment (t4c1) [inline test/unit_test/print_device.cpp:29]:
```suggestion
void print_device_info() {
```

**Review by t4c1 — DISMISSED** [review]

File: test/unit_test/print_device.cpp [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  std::cout << "  Version: " << dev.get_info<info::version>() << "\n";
+  std::cout << "  Driver version: " << dev.get_info<info::driver_version>() << "\n";
+  std::cout << "  Double supported: " << supports_double_str << "\n";
+  std::cout << "  USM supported: " << supports_usm_str << "\n";
+  std::cout << "  Subgroup sizes: " << subgroup_sizes_str << "\n";
+  std::cout << "  Local memory size: " << local_memory_size << "B\n";
+  std::cout << std::endl;
+}
+
+TEST(print_device, run) { print_device(); }
```
[the remark this answers is not in the document]
Comment (Rbiessy) [reply test/unit_test/print_device.cpp]: Renamed this in https://github.com/codeplaysoftware/portFFT/pull/126/commits/f6258487b1aca0e47df80708cb83de4f8c1dc7ad

**Review by t4c1 — APPROVED** [review]

**Review by AD2605 — APPROVED** [review]

**Pull request merged**